### ISSUE TITLE
Multimonitor window location issue (only daemon mode)

### ISSUE THREAD

**Issue by alejandro-perez** [gmc-holle/xfdashboard#88]
Hi,

when I open a window on screen 1, then move it to screen 2, it appears correctly on the side panel, but the miniature is represetned in screen 1. If I move to a different desktop and come back, everything is represented correctly.

The following images depict the issue. If I open thunar on screen 1, move it to screen 2 and then toggle xfdashboard I have the following (incorrect):
![captura de pantalla_2015-04-27_08-43-22](https://cloud.githubusercontent.com/assets/2082536/7342805/8e9d1a28-ecb9-11e4-917a-168944ed8a35.png)
If I move to desktop 4, and come back, I have the following (correct):
![captura de pantalla_2015-04-27_08-43-43](https://cloud.githubusercontent.com/assets/2082536/7342810/98e4929a-ecb9-11e4-8a90-ed610242b9ac.png)

**Comment by gmc-holle** [comment]
Yepp, I see. The reason is that the windows view is listening to "geometry-changed" for windows but only to relayout the current actors in the view. It does check the position of changed window to determine if it is still placed at the monitor where it was before.

I will look deeper at this.

**Comment by gmc-holle** [comment]
I need to implement a new signal 'window-monitor-changed' to get around this issue. Therefore I need to track the last position and size to determine the old monitor where this window was placed. If the new position and size is another monitor I emit this new signal.

The windows view will connect to this signal and determine if it needs to destroy the window actor in its view or the create an actor. This way the window will "move" from one windows view to another and should fix this bug.

**Comment by gmc-holle** [comment]
Should be fixed with commit 5da4858dd005446c50b8fd2c45a1584b1eb6a049

At least it works for me. Can you confirm?

**Comment by alejandro-perez** [comment]
I don't have multimonitor at home. I will test it on Monday.
Thanks!

**Comment by alejandro-perez** [comment]
I'm having problems (CPU 100% ,and stop working) in my multimonitor configuration. It does not happen in my home (single monitor), and it didn't happened before in this computer. It should be related. Could you give me indications to revert to before this commit, and right after, in order to "blame" these modifications?

**Comment by gmc-holle** [comment]
Strange. I don't see why CPU usage shold raise up to 100% and I tested it at home before committing. Does it happen immediately after launching xfdashboard? Or after a window has moved - still staying on the same monitor or placed on another monitor?

I'm not very familiar with git but maybe ```git revert 5da4858dd005446c50b8fd2c45a1584b1eb6a049``` could help to revert only this single changeset.

**Comment by gmc-holle** [comment]
Maybe it is caused by commit bafc9d54a9669e950e3fb58dee9eb8c3dc13c2ba for issue #44 

If ```git revert``` you should also try to rever that commit with ```git revert bafc9d54a9669e950e3fb58dee9eb8c3dc13c2ba```

**Comment by alejandro-perez** [comment]
It's strange. It happens when I move through the workspaces. It happens 
quite fast (just a few seconds).
I will try to revert and will test again.
Thanks!

**Comment by alejandro-perez** [comment]
I will try both separately, and give you some feedback. It might be 
provoked by the recent change on kernel/graphics drivers as well. I 
should not be blaming xfdashboard, actually :D

**Comment by alejandro-perez** [comment]
Definitely, it is this last commit (bafc9d54a9669e950e3fb58dee9eb8c3dc13c2b). I've reverted it back, CPU get's 100% as before when moving a window, but it does not hangs.

(In my rush I forgot to mention that the problem today was not just 100%, but that the computer "hanged" almost completely from the Xorg side. I needed to SSH to the machine and kill xfdashboard to continue working). Reverting this commit everything works as before, but moving windows around get's high CPU load.
